config/helm/aws-node-termination-handler/templates/daemonset.linux.yaml

```diff
@@ -151,6 +151,9 @@ spec:
               value: "false"
             - name: UPTIME_FROM_FILE
               value: {{ .Values.procUptimeFile | quote }}
+            {{- with .Values.extraEnv }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
```

Comment:
When this `extraEnv` block was originally added to the daemonset and deployment templates in #485, it broke out the values into explicit `name` and `value` parameters ([source](https://github.com/aws/aws-node-termination-handler/blob/bef4a565c0efcca22ad916cd2d417c2598a5913e/config/helm/aws-node-termination-handler/templates/daemonset.linux.yaml#L179-L182)):

```yaml
{{- range $key, $value := .Values.extraEnv }}
          - name: {{ $key }}
            value: {{ $value | quote }}
{{- end }}
```

However, during the big refactor in #540, that was changed to the format you're using here ([source](https://github.com/aws/aws-node-termination-handler/blob/main/config/helm/aws-node-termination-handler/templates/deployment.yaml#L167-L169)):

```yaml
            {{- with .Values.extraEnv }}
              {{- toYaml . | nindent 12 }}
            {{- end }}
```

I'm assuming this format is sufficient, since it is working well enough for folks who use NTH as a deployment. Can you confirm from your testing that simply embedding with the `toYaml` directive works correctly for a daemonset?

Reply:
The original method actually accepted a map of env vars like

```
extraEnv:
  FOOVAR: BARVAL
```

even though the default value in `values.yaml` was an empty array.

Technically #540 should've been a breaking change for `extraEnv` even on the `deployment` as it now accepts

```
extraEnv:
  - name: FOOVAR
    value: BARVAL
```

Personally I prefer the map, its less verbose in the values, but that would be a breaking change from the current behaviour!

It does work with the `toYaml` though:

```
> helm template -s templates/daemonset.linux.yaml --set extraEnv[0].name=FOO --set extraEnv[0].value=BAR nth . | grep FOO -A1 -B2
            - name: UPTIME_FROM_FILE
              value: "/proc/uptime"
            - name: FOO
              value: BAR
```

Reply:
You're right, #540 should have been a breaking change for several reasons, including this. I incorrectly tagged it as a patch release (v1.14.1), and after a long back-and-forth (on #565 and #569) we decided to live with the mistake.

Thanks for the explanation here! Glad it's working, even if not ideal.